gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -887,50 +887,59 @@ def test_{{ method_name }}_rest(request_type):
-            result = None
-            is_repeated = False
-            # For repeated fields
-            if isinstance(value, list) and len(value):
-                is_repeated = True
-                result = value[0]
-            # For fields where the type is another message
-            if isinstance(value, dict):
-                result = value
-
-            if result:
-                for nested_key in result.keys():
-                    if (key, nested_key) not in subfield_names:
-                        keys_to_delete.append(
-                            {"key": key, "nested_key": nested_key, "is_repeated": is_repeated}
-                        )
-
-        # Remove fields from the sample request which are not present in the runtime version of the dependency
-        for key_to_delete in keys_to_delete:
-            if key_to_delete.get("nested_key"):
-                if key_to_delete.get("is_repeated"):
-                    del request_init["{{ field.name }}"][key_to_delete.get("key")][0][
-                        key_to_delete.get("nested_key")
-                    ]
-                else:
-                    del request_init["{{ field.name }}"][key_to_delete.get("key")][
-                        key_to_delete.get("nested_key")
+    # Determine if the message type is proto-plus or protobuf
+    is_message_proto_plus_type = not hasattr({{ method.input.ident }}.meta.fields["{{ field.name }}"].message, "DESCRIPTOR")
+
+    if is_message_proto_plus_type:
+        message_fields = {{ method.input.ident }}.meta.fields["{{ field.name }}"].message.meta.fields
+    else:
+        message_fields = {{ method.input.ident }}.meta.fields["{{ field.name }}"].message.DESCRIPTOR.fields
+
+    subfields_not_in_runtime = []
+
+    # Get all subfields for the message
+    nested_fields = [
+        (field.name, subfield.name)
+        for field in message_fields
+        if hasattr(field, "message_type") and field.message_type
+        for subfield in field.message_type.fields
+    ]
+
+    # For each item in the sample request, create a list of sub fields which are not present at runtime
+    for field, value in request_init["{{ field.name }}"].items():
+        result = None
+        is_repeated = False
+        # For repeated fields
+        if isinstance(value, list) and len(value):
+            is_repeated = True
+            result = value[0]
+        # For fields where the type is another message
+        if isinstance(value, dict):
+            result = value
+
+        if result and hasattr(result, "keys"):
+            for subfield in result.keys():
+                if (field, subfield) not in nested_fields:
+                    subfields_not_in_runtime.append(
+                        {"field": field, "subfield": subfield, "is_repeated": is_repeated}
+                    )
+
+    # Remove fields from the sample request which are not present in the runtime version of the dependency
+    for subfield_to_delete in subfields_not_in_runtime:
```

Comment:
To make it easier to read, I suggest factoring out the repeated calls to `get("subfield")`:

```
    for subfield_to_delete in subfields_not_in_runtime:
        field = subfield_to_delete.get("field")
        field_repeated = subfield_to_delete.get("is_repeated")
        subfield = subfield_to_delete.get("subfield"):
        if subfield:
            if field_repeated:
                for i in range(0, len(request_init["{{ field.name }}"][field])):
                    del request_init["{{ field.name }}"][field][i][subfield]
            else:
                del request_init["{{ field.name }}"][field][subfield]
```

Reply:
Fixed in https://github.com/googleapis/gapic-generator-python/pull/1749/commits/402fb17da79e86f6d657ab66efad2da786b946db